### PR TITLE
Increasing maximum number of form fields

### PR DESCRIPTION
**Pull request by smeeckaert**
Hi,

There is a little bottleneck in the application, when you add a big number of fields to one form, you could easily exceed PHP's POST parameters limit (max_input_vars: default 1000). 

To fix the problem i made a modification where instead of sendings fields as POST request, i serialized them in one post field in json, it can handle illimited number of forms until POST max size is reached.

There is still some performance issue, though, with a big number of fields, the ajax request takes ages to process, i will look into it later.

### CODE REVIEW

File: classes/controller/admin/form.ctrl.php [hunk 1]
```diff
@@ -150,7 +155,7 @@ public function action_render_field($item)
         $auto_id = uniqid('auto_id_');
         foreach ($fieldset->field() as $field) {
             if ($field->get_attribute('id') == '') {
-                $field->set_attribute('id', $auto_id.$auto_id_increment++);
+                $field->set_attribute('id', $auto_id . $auto_id_increment++);
```
Comment (jguyomard) [inline classes/controller/admin/form.ctrl.php]: Please remove the spaces around the dot

File: classes/controller/admin/form.ctrl.php [hunk 2]
```diff
@@ -158,14 +163,14 @@ public function action_render_field($item)
             'layout' => $this->config['fields_layout'],
             'fieldset' => $fieldset,
         );
-        $fields_view_params['view_params'] = &$fields_view_params;
+        $fields_view_params['view_params'] = & $fields_view_params;
```
Comment (jguyomard) [inline classes/controller/admin/form.ctrl.php]: No space before &

File: static/js/admin/insert_update.js [hunk 3]
```diff
@@ -59,6 +68,54 @@ define(
...
                 $layout.val(layout);
+
+                var ajaxData = {form_layout: layout};
+                var inputs = {};
+                var fieldPrefix = 'field[';
+                // match field[id][field_name]
+                var idRegex = /[^\[]+\[([0-9]+)\]\[([^\]]+)\]/;
+
+
+                $(this).find('input,textarea').each(function() {
```
Comment (jguyomard) [inline static/js/admin/insert_update.js]: Add a $this = $(this), for better performance

**Comment by jguyomard** [comment]
:+1:

**Comment by Foine** [comment]
:+1:

**Pull request merged**